### PR TITLE
feat(cli): update android template to gradle 8.0

### PR DESCRIPTION
**Pull request by amrbashir**
### What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes, and the changes were approved in issue #___
- [x] No

### Checklist
- [ ] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [x] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).
- [ ] I have added a convincing reason for adding this feature, if necessary

### CODE REVIEW

File: tooling/cli/Cargo.toml [hunk 1]
```diff
@@ -42,7 +42,7 @@ path = "src/main.rs"
 openssl-vendored = [ "tauri-mobile/openssl-vendored" ]
 
 [dependencies]
-tauri-mobile = { version = "0.4", default-features = false }
+tauri-mobile = { git = "https://github.com/tauri-apps/tauri-mobile", default-features = false }
```
Comment (wusyong) [inline tooling/cli/Cargo.toml]: Let's publish tauri-mobile first.

**Review by wusyong — CHANGES_REQUESTED** [review]

File: tooling/cli/Cargo.toml (hunk 1 above)
Comment (amrbashir) [reply tooling/cli/Cargo.toml]: yeah sure, this is mainly to pass in CI and will bring it back before merge and after tauri-mobile is released

**Review by wusyong — APPROVED** [review]

**Review by wusyong — APPROVED** [review]

**Comment by wusyong** [comment]
Android CI seems failed. Can you take a look if you have sometime tmr?

**Comment by amrbashir** [comment]
fixed in #6925

**Comment by amrbashir** [comment]
CI is fixed in `next` branch, I will just go ahead and merge

**Pull request merged**